server/server/data_loader/language_id.py

```diff
@@ -0,0 +1,75 @@
+import subprocess
+import langid
+import tempfile
+import random
+import pathlib
+import lxml.html
+
+def create_model(html_text_path, model_file, sample_size=20*1000*1000):
```

Comment:
this function is super long. Maybe you can extract some code to additional functions